src/libraries/System.Net.WebSockets.Client/tests/WebSocketHelper.cs

```diff
@@ -97,25 +93,48 @@ public static Task<ClientWebSocket> GetConnectedWebSocket(
             Retry(output, async () =>
             {
                 var cws = new ClientWebSocket();
-                configureOptions(cws.Options);
-
-                using (var cts = new CancellationTokenSource(timeOutMilliseconds))
-                {
-                    output.WriteLine("GetConnectedWebSocket: ConnectAsync starting.");
-                    Task taskConnect = invoker == null ? cws.ConnectAsync(server, cts.Token) : cws.ConnectAsync(server, invoker, cts.Token);
-                    Assert.True(
-                        (cws.State == WebSocketState.None) ||
-                        (cws.State == WebSocketState.Connecting) ||
-                        (cws.State == WebSocketState.Open) ||
-                        (cws.State == WebSocketState.Aborted),
-                        "State immediately after ConnectAsync incorrect: " + cws.State);
-                    await taskConnect;
-                    output.WriteLine("GetConnectedWebSocket: ConnectAsync done.");
-                    Assert.Equal(WebSocketState.Open, cws.State);
-                }
+                using var cts = new CancellationTokenSource(timeOutMilliseconds);
+                await ConnectAsync(cws, server, configureOptions, invoker, validateState: true, cts.Token);
                 return cws;
             });
 
+        public static async Task ConnectAsync(
+            ClientWebSocket cws,
+            Uri server,
+            Action<ClientWebSocketOptions> configureOptions,
+            HttpMessageInvoker? invoker = null,
+            bool validateState = true,
+            CancellationToken cancellationToken = default)
+        {
+            if (PlatformDetection.IsNotBrowser && server.Scheme == "wss" && invoker == null)
+            {
+                cws.Options.RemoteCertificateValidationCallback = (_, _, _, _) => true;
+            }
+
+            configureOptions(cws.Options);
+
+            Task taskConnect = invoker == null
+                ? cws.ConnectAsync(server, cancellationToken)
+                : cws.ConnectAsync(server, invoker, cancellationToken);
```

Comment:
```suggestion
            Task taskConnect = cws.ConnectAsync(server, invoker, cancellationToken);
```

Or what am I missing? Cause `ConnectAsync(server, cancellationToken)` seems to be just `ConnectAsync(server, null,  cancellationToken)`.